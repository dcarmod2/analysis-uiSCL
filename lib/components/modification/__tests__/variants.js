import enzyme from 'enzyme'
import React from 'react'

import Variants from '../variants'

test('Components > Modification > Variants', () => {
  const variants = ['a', 'b']
  const setVariant = jest.fn()
  const wrapper = enzyme.shallow(
    <Variants
      allVariants={variants}
      activeVariants={[true, false]}
      modificationId='1'
      setVariant={setVariant}
    />
  )
  expect(wrapper).toMatchSnapshot()
  // Toggle a variant
  const checkBox = wrapper.find({name: 'b'})
  checkBox.simulate('change', {target: {checked: true}})
  expect(setVariant.mock.calls[0][0]).toBe(1)
  expect(setVariant.mock.calls[0][1]).toBe(true)
})
