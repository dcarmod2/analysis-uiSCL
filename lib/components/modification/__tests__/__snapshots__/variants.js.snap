// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > Modification > Variants 1`] = `
<ForwardRef
  spacing={4}
>
  <Heading
    size="md"
  >
    Active in scenarios
  </Heading>
  <ForwardRef
    spacing={2}
  >
    <Checkbox
      fontWeight="normal"
      isChecked={true}
      key="0-a"
      name="a"
      onChange={[Function]}
      value={0}
      wordBreak="break-all"
    >
      1
      . 
      a
    </Checkbox>
    <Checkbox
      fontWeight="normal"
      isChecked={false}
      key="1-b"
      name="b"
      onChange={[Function]}
      value={1}
      wordBreak="break-all"
    >
      2
      . 
      b
    </Checkbox>
  </ForwardRef>
</ForwardRef>
`;
